cli_helpers/tabular_output/preprocessors.py

```diff
@@ -83,6 +83,7 @@ def align_decimals(data, headers, column_types=(), **_):
 
     """
     pointpos = len(headers) * [0]
+    data = list(data)
```

Comment:
Well, converting the database cursor to a list is what I'd like to prevent. And I think it isn't require for all preprocessors to scan through the complete cursor before output can be generated. For example replacement (None -> "NULL") can be done streaming.

I'd like to have some "iterator safe" formatters to parse huge datasets and keep the fancy stuff like nice decimal alignment and automatic column width for the rest of the formatters.